operators/upload_gltf.py

```diff
@@ -0,0 +1,171 @@
+import bpy
+from bpy.types import Operator
+from bpy.props import *
+from os import path
+import requests
+
+from ..globals import (APP_OPERATOR_PREFIX, API_ADDRESS, ION_ADDRESS)
+
+
+def destructure(d, *keys):
+    return [d[k] if k in d else None for k in keys]
+
+
+class S3ProgressPercentage(object):
+    def __init__(self, filename, operator, context):
+        from threading import Lock
+        self._size = float(path.getsize(filename))
+        self._seen_so_far = 0
+        self._lock = Lock()
+        self._context = context
+        self._operator = operator
+
+    def __call__(self, bytes_amount):
+        with self._lock:
+            wm = self._context.window_manager
+            if self._seen_so_far == 0:
+                wm.progress_begin(0, self._size)
+
+            self._seen_so_far += bytes_amount
+            wm.progress_update(self._seen_so_far)
+            percentage = self._seen_so_far / self._size * 100
+            self._operator.report({"INFO"}, f"Progress Upload {percentage}")
+
+            if self._seen_so_far >= self._size:
+                wm.progress_end()
+
+
+class ExportUploadOperator(Operator):
+    bl_label = "gltf to Cesium uploader"
+    bl_idname = f"{APP_OPERATOR_PREFIX}.upload_gltf"
+    bl_description = "Uploads \".glb\" exports to Cesium ion"
```

Comment:
This is oddly phrased.  Let's go with `Uploads your model to Cesium ion for 3D Tiling`